azurelinuxagent/common/protocol/wire.py

```diff
@@ -661,7 +662,13 @@ def fetch_manifest(self, version_uris):
         version_uris_shuffled = version_uris
         random.shuffle(version_uris_shuffled)
 
+        start_time = datetime.now()
         for version in version_uris_shuffled:
+            
+            if datetime.now() - start_time > timedelta(minutes=5):
+                logger.warn("Agent timed-out while fetching extension manifests.")
```

Comment:
the current error message is "Failed to fetch manifest from all sources". Let's improve that for this specific failure to also mention that we timed out after 5 min (maybe including the number of uri that were tried)